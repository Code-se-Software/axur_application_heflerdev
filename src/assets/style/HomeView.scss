.homeview-container {
  > .content {
    min-height: 100vh;
    justify-content: center;
  }

  .container {
    margin: 20px 0;

  }
}

.data-message {
  opacity: 0.4;

  .icon {
    opacity: 0.4;
    display: flex;
    justify-content: center;
  }

  > .text {
    font-size: large;
    font-weight: bolder;
  }

  > p {
    text-align: center;
  }
}