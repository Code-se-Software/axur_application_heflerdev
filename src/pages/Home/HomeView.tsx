import React from "react";
import {Container} from "react-bootstrap";
import {Form, Header, Results} from "./components";

const HomeView = (): any => {
    return (
        <Container>
            <Header/>
            <Form/>
            <Results/>
        </Container>
    )
}

export default HomeView;