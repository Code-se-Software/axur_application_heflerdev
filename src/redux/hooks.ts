import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux'
import type {RootState, AppDispatch} from './store'

export const useAppDispatch: (insertData: { payload: string, type: string }) => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector