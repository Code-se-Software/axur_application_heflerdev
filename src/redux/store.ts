import {configureStore} from '@reduxjs/toolkit'
import websiteReducer from "./slices/websiteSlice"

const store = configureStore({
    reducer: {
        website: websiteReducer,
    },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch