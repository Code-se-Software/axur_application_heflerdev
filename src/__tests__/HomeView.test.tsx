import React from "react";
import TestRenderer from "react-test-renderer";
import {Provider} from "react-redux";
import {store} from "src/redux/store";
import HomeView from "src/pages/Home/HomeView";

describe("<HomeView>", () => {
    const testRenderer = TestRenderer.create(<Provider store={store}><HomeView/></Provider>)
    it("renders correctly", () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
    })
})