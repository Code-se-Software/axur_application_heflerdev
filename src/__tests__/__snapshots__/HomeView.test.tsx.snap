// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView> renders correctly 1`] = `
<div
  className="homeview-container container"
>
  <div
    className="content vstack minBreakpoint-xs"
  >
    <div
      className="justify-content-center row"
    >
      <div
        className="col-12"
      >
        <div
          className="container"
        >
          <div
            className="home-header vstack minBreakpoint-xs"
          >
            <div
              className="icon"
            >
              <span />
            </div>
            <div
              className="text"
            >
              UniCrawler
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-lg-7 col-md-8 col-12"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="vstack gap-3 minBreakpoint-xs"
          >
            <div
              className="search-bar"
            >
              <div
                className="error"
              >
                <span />
              </div>
              <input
                onChange={[Function]}
                placeholder="Need to have 4+ letters."
                type="text"
                value=""
              />
              <div
                className="submit"
                onClick={[Function]}
              >
                <span />
              </div>
            </div>
            <button
              className="submit-button btn btn-primary"
              disabled={false}
              type="submit"
            >
              <span />
            </button>
          </div>
        </form>
      </div>
      <div
        className="container col-12"
      >
        <div
          className="data-message vstack gap-2 minBreakpoint-xs"
        >
          <div
            className="icon"
          >
            <span />
          </div>
          <p
            className="text"
          >
            No content to display
          </p>
          <p>
            To create a new Request, type a keyword on the searchbar above.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
